utils/SystemSetup.ts

```diff
@@ -116,34 +119,57 @@ export async function setUpFunctionalSystem(
     wbtcDecimals = 18;
 
     // swap btc eth price
-    const euroOldInitialWethPrice = euroInitialWethPrice;
-    euroInitialWethPrice = euroInitialWbtcPrice;
-    euroInitialWbtcPrice = euroOldInitialWethPrice;
-
-    const usdOldInitialWethPrice = usdInitialWethPrice;
-    usdInitialWethPrice = usdInitialWbtcPrice;
-    usdInitialWbtcPrice = usdOldInitialWethPrice;
+    const oldInitialWethPrice = initialStablePrice.ETH;
+    initialStablePrice.ETH = initialStablePrice.BTC;
+    initialStablePrice.BTC = oldInitialWethPrice;
   } else {
     wethDecimals = 18;
     wbtcDecimals = 8;
 
-    euroInitialWethPrice = (await getOnChainEthEurPrice(hre)).price;
-    euroInitialWbtcPrice = (await getOnChainBtcEurPrice(hre)).price;
+    // Get stable prices from price feed
+    await Promise.all(
+      getAllBDStablesFiatSymbols().map(async symbol => {
+        const onChainEthStablePrice = symbol === "USD" ? await getOnChainWethUsdPrice(hre) : (await getOnChainEthStablePrice(hre, symbol)).price;
+        _.set(initialStablePrice, ["ETH", symbol], onChainEthStablePrice);
 
-    usdInitialWethPrice = await getOnChainWethUsdPrice(hre);
-    usdInitialWbtcPrice = await getOnChainWbtcUsdPrice(hre);
+        const onChainBtcStablePrice = symbol === "USD" ? await getOnChainWbtcUsdPrice(hre) : (await getOnChainBtcStablePrice(hre, symbol)).price;
+        _.set(initialStablePrice, ["BTC", symbol], onChainBtcStablePrice);
+      })
+    );
   }
 
   if (hre.network.name === "mainnetFork") {
     verboseLog(verbose, "provide liquidity bdus/usdc");
 
     const usdc = (await hre.ethers.getContractAt("IERC20", constants.EXTERNAL_USD_STABLE[hre.network.name].address)) as IERC20;
-    const amount = 100;
-
-    await provideLiquidity(hre, treasury, bdUs, usdc, to_d18(amount), numberToBigNumberFixed(amount, 6), verbose);
-  }
+    const usdAmount = 100;
+
+    await provideLiquidity(hre, treasury, bdUs, usdc, to_d18(usdAmount), numberToBigNumberFixed(usdAmount, 6), verbose);
+
+    verboseLog(verbose, "provide liquidity bxau/weth");
+    const xauValueForLiquidityForPoolSide_bxau_weth = constants.INITIAL_BXAU_UNISWAP_XAU_AMOUNT * scale;
+    await provideLiquidity(
+      hre,
+      treasury,
+      bxau,
+      weth,
+      to_d18(xauValueForLiquidityForPoolSide_bxau_weth),
+      numberToBigNumberFixed(xauValueForLiquidityForPoolSide_bxau_weth, wethDecimals).mul(1e12).div(to_d12(initialStablePrice.ETH.XAU)),
+      verbose
+    );
+
+    verboseLog(verbose, "provide liquidity bgbp/weth");
+    const gbpValueForLiquidityForPoolSide_bgbp_weth = constants.INITIAL_BGBP_UNISWAP_GBP_AMOUNT * scale;
+    await provideLiquidity(
+      hre,
+      treasury,
+      bgbp,
+      weth,
+      to_d18(gbpValueForLiquidityForPoolSide_bgbp_weth),
+      numberToBigNumberFixed(gbpValueForLiquidityForPoolSide_bgbp_weth, wethDecimals).mul(1e12).div(to_d12(initialStablePrice.ETH.GBP)),
+      verbose
+    );
```

Comment:
I did some refactoring here to make it cleaner, but take a look at the calculations we did here around the amounts provided for each LP, it will be difficult to make it generic as there are many different "sub-cases"

Reply:
Yep I get your point